Doesn't compile on clang 18.1

```
[ 50%] Building CXX object CMakeFiles/xunused.dir/main.cpp.o
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++  -I/opt/llvm18/include -I/opt/include -I /opt/openssl-3.2.1/include -O3 -march=native -funroll-loops -finline-functions -mtune=native -Wno-everything -O3 -DNDEBUG -std=gnu++17 -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk -mmacosx-version-min=14.4 -fcolor-diagnostics   -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -MD -MT CMakeFiles/xunused.dir/main.cpp.o -MF CMakeFiles/xunused.dir/main.cpp.o.d -o CMakeFiles/xunused.dir/main.cpp.o -c /xunused/main.cpp
/xunused/main.cpp:178:37: error: no member named 'isPure' in 'clang::CXXMethodDecl'
        if (MD->isVirtual() && !MD->isPure() && MD->size_overridden_methods())
                                ~~  ^
1 error generated.
make[2]: *** [CMakeFiles/xunused.dir/build.make:76: CMakeFiles/xunused.dir/main.cpp.o] Error 1
make[2]: Leaving directory '/xunused'
make[1]: *** [CMakeFiles/Makefile2:218: CMakeFiles/xunused.dir/all] Error 2
make[1]: Leaving directory '/xunused'
make: *** [Makefile:136: all] Error 2

```

--- a/.github/workflows/ccpp.yml
+++ b/.github/workflows/ccpp.yml
@@ -26,6 +26,7 @@ jobs:
 
     - name: Install build prerequisites and Clang/LLVM ${{ matrix.version }}
       run: |
+        sudo apt-get -y remove llvm-14-dev llvm-15-dev
         sudo apt-get -y install build-essential \
                                 cmake \
                                 libcurl4-openssl-dev \

--- a/.github/workflows/ccpp.yml
+++ b/.github/workflows/ccpp.yml
@@ -1,6 +1,9 @@
 name: C/C++ CI
 
-on: [push, pull_request]
+on:
+  push:
+    branches: main
+  pull_request:
 
 jobs:
   llvm:
@@ -9,7 +12,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        version: [13, 14, 15, 16, 17]
+        version: [13, 14, 15, 16, 17, 18]
         os: [ubuntu-22.04]
     
     steps:
@@ -33,6 +36,5 @@ jobs:
     - name: Build xunused for Clang/LLVM ${{ matrix.version }}
       run: |
         mkdir build
-        cd build
-        cmake -DLLVM_VERSION_REQUIRED=${{ matrix.version }} ..
-        make
+        cmake -B build
+        make -C build

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,7 +15,11 @@ message(STATUS "Found LLVM ${LLVM_PACKAGE_VERSION}")
 message(STATUS "Using LLVMConfig.cmake in: ${LLVM_DIR}")
 
 if (LLVM_VERSION_MAJOR GREATER_EQUAL 16)
-    find_package(Clang ${LLVM_VERSION_MAJOR} CONFIG REQUIRED)
+    if (LLVM_VERSION_MAJOR GREATER_EQUAL 18)
+        find_package(Clang ${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR} CONFIG REQUIRED)
+    else()
+        find_package(Clang ${LLVM_VERSION_MAJOR} CONFIG REQUIRED)
+    endif()
     message(STATUS "Using ClangConfig.cmake in: ${Clang_DIR}")
 endif (LLVM_VERSION_MAJOR GREATER_EQUAL 16)
 

--- a/README.md
+++ b/README.md
@@ -4,11 +4,11 @@ It is built upon clang to parse the source code (in parallel). It then shows all
 a definition but no use. Templates, virtual functions, constructors, functions with `static` linkage are
 all taken into account. If you find an issue, please open a issue on https://github.com/mgehre/xunused or file a pull request.
 
-xunused is compatible with LLVM and Clang versions 13 to 17.
+xunused is compatible with LLVM and Clang versions 13 to 18.
 
 ## Building and Installation
 First download or build the necessary versions of LLVM and Clang with development headers.
-On Debian and Ubuntu, this can easily be done via [http://apt.llvm.org](http://apt.llvm.org) and `apt install llvm-17-dev libclang-17-dev`.
+On Debian and Ubuntu, this can easily be done via [http://apt.llvm.org](http://apt.llvm.org) and `apt install llvm-18-dev libclang-18-dev`.
 Then build via
 ```
 mkdir build

--- a/main.cpp
+++ b/main.cpp
@@ -2,6 +2,7 @@
 #include "clang/AST/ASTContext.h"
 #include "clang/ASTMatchers/ASTMatchFinder.h"
 #include "clang/ASTMatchers/ASTMatchers.h"
+#include "clang/Basic/Version.h"
 #include "clang/Driver/Options.h"
 #include "clang/Frontend/ASTConsumers.h"
 #include "clang/Frontend/FrontendActions.h"
@@ -113,13 +114,6 @@ public:
     std::set_difference(Uses.begin(), Uses.end(), Defs.begin(), Defs.end(),
                         std::back_inserter(ExternalUses));
 
-    /*for (auto *F : Uses) {
-        llvm::errs() << "Uses: " << F << " " << F->getNameAsString() << "\n";
-    }
-    for (auto *F : Defs) {
-        llvm::errs() << "Defs: " << F << " " << F->getNameAsString() << "\n";
-    }*/
-
     for (auto *F : ExternalUses) {
       // llvm::errs() << "ExternalUses: " << F->getNameAsString() << "\n";
       std::string USR;
@@ -175,7 +169,13 @@ public:
 
       auto *MD = dyn_cast<CXXMethodDecl>(F);
       if (MD) {
-        if (MD->isVirtual() && !MD->isPure() && MD->size_overridden_methods())
+        if (MD->isVirtual()
+        #if CLANG_VERSION_MAJOR >= 18
+          && !MD->isPureVirtual()
+        #else
+          && !MD->isPure()
+        #endif
+          && MD->size_overridden_methods())
           return; // overriding method
         if (isa<CXXDestructorDecl>(MD))
           return; // We don't see uses of destructors.

